### PR TITLE
[CSL-1123] Add new scripts in 0.5-rc branch.

### PR DESCRIPTION
**Pull request by denisshevchenko**

### CODE REVIEW

File: scripts/haskell/update-cabal-versions.sh [hunk 1]
```diff
@@ -16,6 +16,7 @@ updateVersion core/cardano-sl-core.cabal
 updateVersion infra/cardano-sl-infra.cabal
 updateVersion db/cardano-sl-db.cabal
 updateVersion lrc/cardano-sl-lrc.cabal
+updateVersion lwallet/cardano-sl-lwallet.cabal
```
Comment (gromakovsky) [inline scripts/haskell/update-cabal-versions.sh]: Wait, there is no `lwallet` in 0.5-rc.

File: scripts/haskell/lint.sh [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env bash
-projects="core db lrc infra update ssc godtossing txp"
+projects="core db lrc infra update ssc godtossing txp lwallet"
```
Comment (gromakovsky) [inline scripts/haskell/lint.sh]: There is no `lwallet`.

File: scripts/grep.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# This utility script is written by @gromak and is provided "as is".
+# The author doesn't guarantee anything about it. It might work.
+
+grep --color=auto -r "$@" src test bench core/Pos update/Pos db/Pos lrc/Pos infra/Pos ssc/Pos godtossing/Pos txp/Pos lwallet/Pos lwallet/*.hs
```
Comment (gromakovsky) [inline scripts/grep.sh]: `lwallet`?

File: scripts/build/cardano-sl.sh [hunk 4]
```diff
@@ -169,9 +172,13 @@ if [[ $spec_prj == "" ]]; then
   for prj in $projects; do
     to_build="$to_build cardano-sl-$prj"
   done
-  to_build="$to_build cardano-sl"
+  to_build="$to_build cardano-sl cardano-sl-lwallet"
```
Comment (gromakovsky) [inline scripts/build/cardano-sl.sh]: No `lwallet`.

File: scripts/build/cardano-sl.sh [hunk 5]
```diff
@@ -169,9 +172,13 @@ if [[ $spec_prj == "" ]]; then
...
 elif [[ $spec_prj == "sl" ]]; then
   to_build="cardano-sl"
+elif [[ $spec_prj == "lwallet" ]]; then
```
Comment (gromakovsky) [inline scripts/build/cardano-sl.sh]: `lwallet`?

File: scripts/build/cardano-sl.sh [hunk 6]
```diff
@@ -169,9 +172,13 @@ if [[ $spec_prj == "" ]]; then
...
+  to_build="cardano-sl-lwallet"
+elif [[ $spec_prj == "sl+" ]]; then
+  to_build="cardano-sl cardano-sl-lwallet"
```
Comment (gromakovsky) [inline scripts/build/cardano-sl.sh]: Why `lwallet`?

File: scripts/build/cardano-sl.sh [hunk 7]
```diff
@@ -32,8 +41,8 @@ set -o pipefail
 # * Pass --no-nix or do `touch .no-nix` if you want builds without Nix
 # * Pass --ram or do `touch .ram`. if you have lots of RAM and want to
 #   make builds faster
-# * Pass -Werror or do `touch .Werror` if you want to compile with -Werror
 
+# We can't have lwallet here, because it depends on 'cardano-sl'.
```
Comment (gromakovsky) [inline scripts/build/cardano-sl.sh]: And here.

File: scripts/build/cardano-sl.sh [hunk 8]
```diff
@@ -32,8 +41,8 @@ set -o pipefail
 # * Pass --no-nix or do `touch .no-nix` if you want builds without Nix
 # * Pass --ram or do `touch .ram`. if you have lots of RAM and want to
 #   make builds faster
-# * Pass -Werror or do `touch .Werror` if you want to compile with -Werror
```
Comment (gromakovsky) [inline scripts/build/cardano-sl.sh]: And this is bad.

File: scripts/build/cardano-sl.sh [hunk 9]
```diff
@@ -49,17 +58,13 @@ prod=false
 wallet=true
 explorer=false
 no_code=false
-werror=false
```
Comment (gromakovsky) [inline scripts/build/cardano-sl.sh]: This is bad too.

File: scripts/build/cardano-sl.sh [hunk 10]
```diff
@@ -49,17 +58,13 @@ prod=false
...
 
 if [ -e .no-nix ]; then
   no_nix=true
 fi
 if [ -e .ram ]; then
   ram=true
 fi
-if [ -e .Werror ]; then
-  werror=true
```
Comment (gromakovsky) [inline scripts/build/cardano-sl.sh]: And here.

File: scripts/build/cardano-sl.sh [hunk 11]
```diff
@@ -78,16 +83,13 @@ do
   # --ram = use more RAM
   elif [[ $var == "--ram" ]]; then
     ram=true
-  # -Werror = compile with -Werror
-  elif [[ $var == "-Werror" ]]; then
-    werror=true
```
Comment (gromakovsky) [inline scripts/build/cardano-sl.sh]: And so on.

File: scripts/ci/travis.sh [hunk 12]
```diff
@@ -38,16 +38,22 @@ if [[ "$with_haddock" == "true" ]]; then
...
+
+for trgt in $targets; do
+
+    stack --nix --no-terminal --local-bin-path daedalus/ install "$trgt" \
+      $EXTRA_STACK --fast --jobs=2 \
+      --ghc-options="-j -DCONFIG=wallet +RTS -A128m -n2m -RTS" \
+      --flag cardano-sl-core:-asserts \
+      --flag cardano-sl-core:-dev-mode
+
+done
```
Comment (gromakovsky) [inline scripts/ci/travis.sh]: This change is irrelevant too.

**Comment by gromakovsky** [comment]
Perhaps you didn't cherry-pick, but did something hacky instead.

File: appveyor.yml [hunk 13]
```diff
@@ -75,7 +75,8 @@ test_script:
       --extra-include-dirs="%WORK_DIR%\rocksdb\include"
       --extra-lib-dirs="%WORK_DIR%"
   # Retry transient failures due to https://github.com/haskell/cabal/issues/4005
-  - scripts\appveyor-retry call stack install cardano-sl
+  # We intentionally don't build lwallet here, because this build is for installer.
```
Comment (gromakovsky) [inline appveyor.yml]: This line doesn't make sense in this branch as there is no `lwallet`.

**Comment by denisshevchenko** [comment]
Removed `lwallet`.

**Pull request merged**